ckanext/xloader/helpers.py

```diff
@@ -42,3 +44,82 @@ def is_resource_supported_by_xloader(res_dict, check_access=True):
     else:
         is_supported_url_type = True
     return (is_supported_format or is_datastore_active) and user_has_access and is_supported_url_type
+
+
+def xloader_badge(resource):
+    # type: (dict) -> str
+    """
+    Displays a custom badge for the status of Xloader and DataStore for the specified resource.
+    """
+    if not toolkit.asbool(toolkit.config.get('ckanext.xloader.show_badges', True)):
+        return ''
+
+    if not XLoaderFormats.is_it_an_xloader_format(resource.get('format')):
+        # we only want to show badges for supported xloader formats
+        return ''
+
+    is_datastore_active = resource.get('datastore_active', False)
+
+    try:
+        xloader_job = toolkit.get_action("xloader_status")({'ignore_auth': True},
+                                                           {"resource_id": resource.get('id')})
+    except toolkit.ObjectNotFound:
+        xloader_job = {}
+
+    if xloader_job.get('status') == 'complete':
```

Comment:
Could refactor this to extract `xloader_job.get('status')` to a variable at the start.

Reply:
Is this required or just a nice code clean up @ThrawnCA .

Reply:
Just cleanup.